pubsub/topics.rb

```diff
@@ -286,6 +303,64 @@ def publish_messages_async_with_concurrency_control project_id:, topic_name:
   # [END pubsub_publisher_concurrency_control]
 end
 
+def publish_ordered_messages project_id:, topic_name:
+  # [START pubsub_publish_ordered_messages]
+  # project_id = "Your Google Cloud Project ID"
+  # topic_name = "Your Pubsub topic name"
+  require "google/cloud/pubsub"
+
+  pubsub = Google::Cloud::Pubsub.new project: project_id
+
+  # Start sending messages in one request once the size of all queued messages
+  # reaches 1 MB or the number of queued messages reaches 20
+  topic = pubsub.topic topic_name, async: {
+    max_bytes:    1_000_000,
+    max_messages: 20
+  }
+  topic.enable_message_ordering!
+  10.times do |i|
+    topic.publish_async "This is message \##{i}.",
+                        ordering_key: "ordering-key"
+  end
+
+  # Stop the async_publisher to send all queued messages immediately.
+  topic.async_publisher.stop.wait!
+  puts "Messages published asynchronously in batch."
+  # [END pubsub_publish_ordered_messages]
+end
+
+def publish_resume_publish project_id:, topic_name:
+  # [START pubsub_publish_resume_publish]
+  # project_id = "Your Google Cloud Project ID"
+  # topic_name = "Your Pubsub topic name"
+  require "google/cloud/pubsub"
+
+  pubsub = Google::Cloud::Pubsub.new project: project_id
+
+  # Start sending messages in one request once the size of all queued messages
+  # reaches 1 MB or the number of queued messages reaches 20
+  topic = pubsub.topic topic_name, async: {
+    max_bytes:    1_000_000,
+    max_messages: 20
+  }
+  topic.enable_message_ordering!
+  10.times do |i|
+    topic.publish_async "This is message \##{i}.",
+                        ordering_key: "ordering-key" do |result|
+      if result.succeeded?
+        puts "Message \##{i} successfully published."
+      else
+        puts "Message \##{i} failed to publish"
+        topic.resume_publish "ordering-key"
+      end
+    end
+  end
+
+  # Stop the async_publisher to send all queued messages immediately.
+  topic.async_publisher.stop.wait!
```

Comment:
[AsyncPublisher#stop!](https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-pubsub/lib/google/cloud/pubsub/async_publisher.rb#L183-L197) combines these two methods, if you would prefer to show it instead. In google-cloud-ruby's inline api samples, `stop!` is shown more than `stop` and `wait!`, but there are still 15 usages of the latter, so I suppose either is OK for now.